x-pack/plugins/profiling/server/lib/setup/get_has_setup_privileges.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from '@kbn/core/server';
+import { INTEGRATIONS_PLUGIN_ID, PLUGIN_ID as FLEET_PLUGIN_ID } from '@kbn/fleet-plugin/common';
+import { ProfilingPluginStartDeps } from '../../types';
+
+export async function getHasSetupPrivileges({
+  securityPluginStart,
+  request,
+}: {
+  securityPluginStart: NonNullable<ProfilingPluginStartDeps['security']>;
+  request: KibanaRequest;
+}) {
+  const { hasAllRequested } = await securityPluginStart.authz
```

Comment:
I failed to mention that you first need to determine if you should check privileges in the first place:

```ts
if (securityPluginStart.authz.mode.useRbacForRequest(request)) {
  // Perform privileges check
} else {
  // You can safely bypass the privilege check
}